### PR TITLE
remove ffmpeg, openh264 and update qt to 5.10

### PR DESCRIPTION
**Pull request by swflb**

### CODE REVIEW

File: Findscript.cmake.in [hunk 1]
```diff
@@ -1,11 +1,11 @@
-# - Find an ovsrpro installation.
-# ovsrpro_DIR
+# - Find an @CMAKE_PROJECT_NAME@ installation.
+# @CMAKE_PROJECT_NAME@_DIR
```
Comment (smanders) [inline Findscript.cmake.in]: (question) is there a reason for ovsrpro to have it's own version of `Findscript.cmake.in`? all our other `*pro` repos (internpro and webpro) don't need to do this - they use the Findscript.cmake.in from externpro...

if you can help me identify what you want to customize then perhaps I can add these things to the next release of externpro...

File: Findscript.cmake.in [hunk 2]
```diff
@@ -75,58 +75,54 @@ function(getNumBits _ret)
...
-    "/opt/ovsrpro/ovsrpro-${ovsrpro_SIG}-${CMAKE_SYSTEM_NAME}"
-    # symbolic link install option
-    "/opt/ovsrpro/ovsrpro"
-    # rpm installed location
-    "/opt/ovsrpro"
-  DOC "ovsrpro directory"
+    "$ENV{ProgramW6432}/@CMAKE_PROJECT_NAME@ ${@CMAKE_PROJECT_NAME@_SIG}"
+    "$ENV{${PFX86}}/@CMAKE_PROJECT_NAME@ ${@CMAKE_PROJECT_NAME@_SIG}"
+    "~/ovsrpro/@CMAKE_PROJECT_NAME@-${@CMAKE_PROJECT_NAME@_SIG}-${CMAKE_SYSTEM_NAME}"
+    "/opt/ovsrpro/@CMAKE_PROJECT_NAME@-${@CMAKE_PROJECT_NAME@_SIG}-${CMAKE_SYSTEM_NAME}"
```
Comment (smanders) [inline Findscript.cmake.in]: does the install path on UNIX systems need to be `~/ovsrpro` and `/opt/ovsrpro`? the recent change in externpro to make this `~/extern` and `/opt/extern` was meant to be a more generic place to install - and a single mount point on our shared development systems (install once on a single system and have externpro/internpro/webpro and skylynx pro-like installs all in `/opt/extern`)

File: CMakeLists.txt [hunk 3]
```diff
@@ -28,7 +28,7 @@ proExecuteStep()
 
 if(${XP_STEP} STREQUAL "build")
 # move some needed cmake files into place
-configure_file(Findovsrpro.cmake ${STAGE_DIR}/share/cmake/Findovsrpro.cmake COPYONLY)
+#configure_file(Findovsrpro.cmake ${STAGE_DIR}/share/cmake/Findovsrpro.cmake COPYONLY)
```
Comment (smanders) [inline CMakeLists.txt]: (question) I'm guessing that since you've commented this out and renamed `Findovsrpro.cmake` to `Findscript.cmake.in` that this line could just be removed?

File: projects/qt5.cmake [hunk 4]
```diff
@@ -23,9 +23,9 @@ set(PRO_QT5
   VER ${QT5_VER}
   GIT_ORIGIN ${QT5_REPO}
   GIT_TAG v${QT5_VER}
-  DLURL http://download.qt.io/archive/qt/5.8/${QT5_VER}/single/${QT5_DOWNLOAD_FILE}
-  DLMD5 a9f2494f75f966e2f22358ec367d8f41
-  PATCH ${PATCH_DIR}/qt5.patch
+  DLURL http://download.qt.io/archive/qt/5.10/${QT5_VER}/single/${QT5_DOWNLOAD_FILE}
+  DLMD5 c5e275ab0ed7ee61d0f4b82cd471770d
+#  PATCH ${PATCH_DIR}/qt5.patch
```
Comment (smanders) [inline projects/qt5.cmake]: (optional) remove commented-out line?

**Review by smanders — APPROVED** [review]

File: Findscript.cmake.in (hunk 1 above)
Comment (swflb) [reply Findscript.cmake.in]: @smanders the reason we assumed we need a version here is that it does not have the same release cycle as externpro. so we could use externpro 17.05.1 (just example) and ovspro 17.10.2. Is there a better way?

File: Findscript.cmake.in (hunk 2 above)
Comment (swflb) [reply Findscript.cmake.in]: @smanders that would be helpful. it will of course cause some downstream changes but that would be excellent. do we just need to remove this Findscript file?

File: Findscript.cmake.in (hunk 1 above)
Comment (smanders) [reply Findscript.cmake.in]: I'm not seeing how the release cycle has anything to do with this file...
* ovsrpro specifies the release of externpro it's using
  * https://github.com/distributePro/ovsrpro/blob/master/CMakeLists.txt#L4
* the `Findscript.cmake.in` from that release of externpro would be used to generate the `Findovsrpro.cmake` file that's in the ovsrpro installer and copied into projects that require (want to find) ovsrpro...

what am I missing? perhaps you could explain the release cycle thought? are you wanting to update the `Findovsrpro.cmake` to support a new release of Visual Studio, for example, without specifying the release of externpro that supports the new release of Visual Studio?

Comment (smanders) [reply Findscript.cmake.in]: Using your example (use externpro 17.05.1 and ovsrpro 17.10.2)... are you saying that there need to be additional things in `Findovsrpro.cmake` 17.10.2 that aren't in the `Findscript.cmake.in` from externpro 17.05.1? again - I think we need to identify what things in `Findscript.cmake.in` that you want to customize

File: Findscript.cmake.in (hunk 2 above)
Comment (smanders) [reply Findscript.cmake.in]: before we remove it I want to make sure the externpro `Findscript.cmake.in` has everything you want/need

Comment (swflb) [reply Findscript.cmake.in]: @smanders we were also using the old Findovspro.cmake file just like the externpro was used by code that included this stuff. if we don't need the find script anymore, that is fine by me.

File: Findscript.cmake.in (hunk 1 above)
Comment (swflb) [reply Findscript.cmake.in]: @smanders we don't really need to customize stuff in here. we think/thought we needed it to specify install location (old to /opt/ovsrpro but new looks like /opt/extern/ovspro. right?), to specify version and to find the ovsrpro libraries. if all that is not needed, then i am for removing this.

File: Findscript.cmake.in (hunk 2 above)
Comment (smanders) [reply Findscript.cmake.in]: Maybe there’s just confusion on what the Find*pro.cmake script is for...

* externpro does not have a Findexternpro.cmake file in it’s repo
* the Findexternpro.cmake is generated at cmake-time from the Findscript.cmake.in template
* the generated Findexternpro.cmake file is in the externpro install (share/cmake)
* this generated file is copied to projects that need to find externpro
* the project finds externpro via this file and compares it to the Findexternpro.cmake that’s in the release of externpro that it found — if they’re different then the developer gets a cmake warning that the files don’t match and the updated version from externpro should be copied to the project

bottom line is that you shouldn’t need to have a Findovsrpro.cmake in the ovsrpro repo - it will be generated at cmake-time and will be part of the install (in share/cmake)

and if the Findscript.cmake.in template (from externpro) doesn’t have something you need in your Findovsrpro.cmake then we should consider modifying Findscript.cmake.in in externpro...

Comment (swflb) [reply Findscript.cmake.in]: @smanders thanks for the explanation. that helps a lot. I will remove the find script from here.

Comment (swflb) [reply Findscript.cmake.in]: @smanders is there a macro i need to call to ensure the findovspro script is generated?

File: Findscript.cmake.in (hunk 1 above)
Comment (smanders) [reply Findscript.cmake.in]: cmake looks in `/opt/extern` for an externpro or internpro or webpro or ovsrpro install... (it used to look in `/opt/externpro` for an externpro or internpro or webpro install)

install command and directory listing:
```bash
$ sudo ./externpro-18.01.1-gcc630-64-Linux.sh --prefix=/opt/extern --include-subdirs
$ ls -1 /opt/extern
externpro-18.01.1-gcc630-64-Linux
externpro-18.01.1-gcc720-64-Linux
internpro-18.01.2-gcc630-64-Linux
webpro-18.01.1-gcc630-64-Linux
webpro-SL18.01.1-gcc630-64-Linux
```

File: Findscript.cmake.in (hunk 2 above)
Comment (smanders) [reply Findscript.cmake.in]: @swflb no, you shouldn’t need to call a macro - it should be done for you...
```bash
$ cmake -DXP_STEP=build /path/to/src
```
look in staging directory share/cmake (for example `internpro_18.01.2-gcc630-64/share/cmake/` in the build directory for internpro) and you should see a Find*pro.cmake (Findinternpro.cmake in internpro)

how this happens:
* these lines are called from `proSetStageDir()` macro https://github.com/smanders/externpro/blob/18.01.1/modules/macpro.cmake#L292-L295
* the `proSetStageDir()` macro is called from the cmake-generated pro_build.cmake file that’s in your build directory https://github.com/smanders/externpro/blob/18.01.1/modules/macpro.cmake#L245

https://github.com/smanders/externpro/blob/18.01.1/modules/macpro.cmake#L142

**Comment by brobeson** [comment]
@swflb Ready to merge this one?

File: CMakeLists.txt [hunk 5]
```diff
@@ -28,7 +28,6 @@ proExecuteStep()
 
 if(${XP_STEP} STREQUAL "build")
 # move some needed cmake files into place
-configure_file(Findovsrpro.cmake ${STAGE_DIR}/share/cmake/Findovsrpro.cmake COPYONLY)
```
Comment (tjflexmaster) [inline CMakeLists.txt:31]: Shouldn't this be 
```configure_file(/opt/extern/externpro/Findscript.cmake.in ${STAGE_DIR}/share/cmake/Findovsrpro.cmake @ONLY)```

Comment (smanders) [reply CMakeLists.txt:31]: no - he's deleting this line

also - commenting on your suggestion: you would never want a hard-coded path like `/opt/extern` (the rest of the path isn't correct, either) - you want to "find" the installed externpro and use the script from share/cmake

I attempted to explain how the script is created in comments in this pull (outdated and hidden now) https://github.com/distributePro/ovsrpro/pull/30#discussion_r163688291

Comment (tjflexmaster) [reply CMakeLists.txt:31]: Ahh, I understand now.   Thanks.

@smanders  What is the advantage of copying this file instead of adding the install path to directories searched by find_package?

Comment (smanders) [reply CMakeLists.txt:31]: @tjflexmaster maybe I don't understand your question - are you asking if you could replace `Findscript.cmake.in` (which generates `Findovsrpro.cmake`) with `find_package`? this file (`Findscript.cmake.in`) does a lot more than `find_package` -- but I suppose you could replace it if you wanted to add the complications to the project that uses the *pro project...

Comment (tjflexmaster) [reply CMakeLists.txt:31]: @smanders Let me try again.  If I call find_package(ovsrpro MODULE REQUIRED) in my cmake it can find and load the Findovsrpro.cmake that is installed when ovsrpro was installed (/opt/extern/ovsrpro/share/cmake/Findovsrpro.cmake).  All I have to do is add /opt/extern/ovsrpro... to the CMAKE_MODULE_PATH list.  I am not talking about replacing Findscript.cmake.in I am talking about using the file generated by the build directly instead of using a copy of it.

**Review by bnhhughes — APPROVED** [review]
@swflb I take it you got the newer Qt to build correctly now? Some good cleanup too.

File: CMakeLists.txt (hunk 5 above)
Comment (smanders) [reply CMakeLists.txt:31]: @tjflexmaster you don't like the copy? is there something difficult about it? or just trying to simplify?

I believe your simplifying assumption is that you only look in `/opt/extern` -- but you also need the ovsrpro directory with all it's compiler, bits, and OS decorations -- the `Findscript.cmake.in` supports multiple locations to look for it: developer "build" versions, using an environment variable, and multiple install locations -- Windows-specific and UNIX-specific

would you suggest adding all of those locations to your `CMAKE_MODULE_PATH`?

you are correct -- if you only support a single location to find ovsrpro then you could skip the copy to your project that uses ovsrpro and add the install location to `CMAKE_MODULE_PATH` so `find_package()` could find it

Comment (tjflexmaster) [reply CMakeLists.txt:31]: @smanders The copy isn't a big deal.  I was just wondering about the reasons.  I typically only have ovsrpro installed in one location so telling cmake what that is through a cmake variable at configuration time or through environment variables is easy enough.  But with the number of developers and the different number of environments you need to support I can see the benefit of taking this duty off of the developer.

Comment (smanders) [reply CMakeLists.txt:31]: @tjflexmaster it's always good to ask the questions... sometimes it will make me rethink why I did do things the way I did -- so many of these things were done in my early days of cmake, too -- so I won't be the least bit surprised if there are better ways to do some of the things I've done -- but I have put a lot of thought and trial and error into what there is in externpro today (over the years)

**Comment by swflb** [comment]
@brobeson I just merged the other recent changes, so let me build check this. then it should be ready to merge.

**Comment by swflb** [comment]
@brobeson at least as far as I know. did I miss any requested changes?

**Comment by brobeson** [comment]
@swflb 

>did I miss any requested changes?

Not that I'm aware of. I was checking if you had any new commits to push for any reason before I merged. Since you're rebuilding, I'll let you handle merging this after you confirm.
